templates/glossary.md

```diff
@@ -0,0 +1,331 @@
+It consists of both fully automated generation of Lean 4 [olean files](#olean-file) from Lean 3 source files, as well as best-effort source-to-source translation of Lean 3 to Lean 4 source code.
+Learnings from both the mathport effort as well as from [mathlib4](#mathlib4) often lead to backported changes to mathlib, to bring Lean 3 code into more future-compatible states.
+
+### mode
+
+In the context of writing Lean code, a set of related syntactical elements or keywords which make a particular style of formal reasoning or construction of [proof terms](#proof-term) efficient.
+Lean, especially as restricted to [mathlib](#mathlib), has a small number of such modes -- [tactic mode](#tactic-mode), [term mode](#term-mode), [calc mode](#calc-mode) and [conv-mode](#conv-mode).
+A particular mode may make progress towards specific kinds of [goals](#goal) easier.
+A proof may often, however, mix various modes in the course of constructing a [proof term](#proof-term).
+
+### module
+
+A single file containing Lean source code.
+
+Not to be mistaken with a `module` within mathematics, i.e. the generalization of a vector space.
+
+## module docstring
+
+[Module](#module)-level comment summarizing what's to be found in the file.
+We require that every file has one, but [some old files](https://github.com/leanprover-community/mathlib/blob/master/scripts/style-exceptions.txt) still don't.
+
+### MWE
+
+*Minimal Working Example*, a way of making it easier to get help with a snippet of Lean code by reducing it to its essential parts, whilst being still runnable by others.
+
+Further information can be found on [the MWE page](mwe.html).
+
+### non-terminal `simp`
+
+An invocation of the `simp` tactic which is not the last tactic invoked on a particular [subgoal](#goal), nor uses `simp only` to explicitly limit which [simp lemmas](#simp-lemma) it considers.
+Non-terminal `simp`s are avoided because they are hard to maintain, given that their behavior or runtime will change as mathlib adds or modifies the set of `simp` lemmas over time.
+
+#### See also
+
+The ["non-terminal `simp`" section](https://leanprover-community.github.io/extras/simp.html#non-terminal-codesimpcodes) of the `simp` documentation
+
+### olean file
+
+Lean code has to be compiled.
+The compiled version of file `x.lean` is file `x.olean` and all olean files together form the [cache](#cache).
+
+### orange bar of hell
+
+In VScode, the *orange bar of hell* refers to the orange bar that appears left to the current file when it persists.
+The reason is that the Lean extension has to (re)compile all the imported files whose [cache](#cache) does not match.
+Because of this, having branches that touch two files far apart (with respect to the [import tree](#import-tree)) related through imports is considered bad practice as **any** modification on the file upstream will force Lean to recompile all the files in the middle.
+
```

Comment:
How about adding suggestions here for escaping hell? I give my students this snippet which could be adapted:

Lean in VSCode is usually pretty responsive. If it's feeling slow, or you're seeing orange lines on the side that aren't going away, there are a few things to try.

First, in VSCode, close any editor tabs that you aren't using anymore. Then open the Command Palette (ctrl-shift-p or cmd-shift-p) and run Lean: Restart. You might find it useful to map this to a hotkey in VSCode. I use ctrl-r.

If that doesn't work, let's make sure you have a fresh copy of the library. In the root fpv2021 directory, run:

```
leanpkg configure
leanproject get-mathlib
lean --make src/lovelib.lean
```

Reply:
Done, thanks!